### PR TITLE
[WIP] Mass convert of *^Uf terrain to the balance equivalent Tb^Tf terrain.

### PR DESCRIPTION
**Pull request by Pentarctagon**
---

This is a mass-conversion of all remaining *^Uf terrain to Tb^Tf, since if we can't make sure that something that's deprecated isn't used by mainline anymore then it shouldn't have been deprecated in the first place.  It does not touch WC, since I assume @gfgtdf has his own plans to handle WC (#5135).  Marked WIP since I also noticed along the way that the Xol and Xuce terrains are deprecated but also still used in a few places, so I intend to review all the deprecated terrains and make sure they aren't used in any mainline maps anymore.

@gfgtdf I've requested your review as well to sanity check the changes to `data/multiplayer/scenarios/test_lua_mapgen.lua` in particular.

### CODE REVIEW

**Comment by stevecotton** [comment]
While it will be a lot of effort, I'd prefer this to be done with a manual review of each map, rather than an bulk search-and-replace. Some of the ^Uf tiles should be Tb^Tf, but others would be better as Uu^Tf or Hh^Tf. Leaving the existing ^Uf tiles as-is both tells us which maps haven't been reviewed yet, and during a map review helps track which hexes have already been considered.

The evil path of Northern Rebirth should probably be removed, rather than being part of this.

**Comment by stevecotton** [comment]
For Xuce, the only instances are some leftovers (that aren't visible to the player) in HttT's 14_Plunging_Into_the_Darkness.map . That also has ^Uf mushrooms, but only in a secret area.

File: data/campaigns/Under_the_Burning_Suns/utils/animation_macros.cfg [hunk 1]
```diff
@@ -22,7 +22,5 @@ misc/blank-hex.png~BLIT(units/quenoth/fire/flame-color.png,{X},{Y2})~MASK(units/
 
 # Not all unwalkable, but still supposed to provide some visual cue that movement is different.
 # This is only used for animation, it's the list of terrains on which the Sun Sylph will fly.
-# Both the old ^Uf* and new ^Tf* mushrooms are supported here - if the Quenoth elves are moved
-# to core then it seems sensible to still have the old behavior on old maps.
```
Comment (stevecotton) [inline data/campaigns/Under_the_Burning_Suns/utils/animation_macros.cfg:26]: I think this should stay in, it applies to old UMC maps even when there are no ^Ufs in mainline.

**Comment by gfgtdf** [comment]
data/multiplayer/scenarios/test_lua_mapgen.lua is basciaiyll the WC2 maritime mapgenerator in a one-file version, in fact i consider removing it now that WC2 is mainlined.

File: data/campaigns/Under_the_Burning_Suns/utils/animation_macros.cfg (hunk 1 above)
Comment (Pentarctagon) [reply data/campaigns/Under_the_Burning_Suns/utils/animation_macros.cfg:26]: Alright, I'll push an update.

**Comment by Pentarctagon** [comment]
@gfgtdf is there anything that still uses it, or can I delete it here?

**Comment by gfgtdf** [comment]
> @gfgtdf is there anything that still uses it, or can I delete it here?

No it's a test as the name says, it was never used by any other code.

**Comment by gfgtdf** [comment]
> While it will be a lot of effort, I'd prefer this to be done with a manual review of each map

I agree on this part, in particular since i think on a lot of these  of these maps the balance effect of these is negligible,  So it'd probably be better to use the old graphocs that the scenario designer had in mind in the first place.

**Comment by Pentarctagon** [comment]
I'm fine with someone manually reviewing the maps as long as it's done (read: all *^Uf is removed from mainline maps) prior to 1.16's release.  Which would by default mean @nemaara for SP and then probably @Hejnewar et al for MP.

**Comment by Hejnewar** [comment]
Yeah, I think it's better to do this this way, . It's gonna take some work and time but better looks (hopefully) will be worth it.

**Comment by soliton-** [comment]
I think in campaigns the chance is low that the mixed mushroom terrain would break something. It's likely used primarily for the aesthetics.

For the balanced multiplayer maps on the other hand it likely would since it is often used precisely for its gameplay properties that the mixed terrain does not have.

**Comment by Pentarctagon** [comment]
@nemaara then - will you be able to do the review of *^Uf terrain usage in SP maps?

**Comment by nemaara** [comment]
I'm confused, does this PR leave the tiles looking the same as before, or do they look different? If it's the same look then SP doesn't need a manual review and a mass change is fine.

**Comment by Pentarctagon** [comment]
The look is the same as it was previously in 1.15, but different from 1.14.  For example, the Slavers map from DW-

1.14 (Steam):
DW-Slavers-1 14-Steam
1.15 (Steam):
DW-Slavers-1 15-Steam
1.15 (With PR applied):
DW-Slavers-1 15-With-PR

**Comment by soliton-** [comment]
Mass conversion for SP should use the mixed terrain that keeps the aesthetics not the gameplay properties.

**Comment by Pentarctagon** [comment]
Ultimately which way to handle it is up to nemaara.  The only restriction I'm placing on it though is that, one way or another, the deprecated mushroom terrain won't be used by mainline maps in 1.16.

**Comment by nemaara** [comment]
I agree with Soliton. Let's keep the mushrooms with the same look as 1.14.

**Comment by Pentarctagon** [comment]
Alright, will update.

**Comment by Pentarctagon** [comment]
@nemaara I've changed the SP maps to use the overlay instead.

@gfgtdf I went ahead and deleted the test_lua_mapgen.lua and test_lua_mapgen.cfg files.

@Hejnewar I've opened #5145 for the MP maps that still use `^Uf`.

File: data/multiplayer/scenarios/2p_Elensefar_Courtyard.cfg [hunk 2]
```diff
@@ -96,7 +96,7 @@
             [/or]
             [or]
                 x,y=32-36,9-19
-                terrain=R*,Gg^Emf,Rp^Uf,Rp^Dr,Rd^Uf
+                terrain=R*,Gg^Emf,Tb^Tf,Rp^Dr
```
Comment (stevecotton) [inline data/multiplayer/scenarios/2p_Elensefar_Courtyard.cfg:99]: This needs to be in #5145 , along with the change to the corresponding map.

File: data/multiplayer/scenarios/ANL_utils/ANL_worker_options.cfg [hunk 3]
```diff
@@ -473,7 +473,7 @@
                             image = "terrain/cave/floor.png"
                             label = "<span color='green'>" + _ "Harvest Mushrooms" + "</span>"
                             description = "<span size='small'>" + _ "Earns: 3g" + "</span>"
-                            {ANL_SHOW_IF ("Peasant") "Uu^Uf"}
+                            {ANL_SHOW_IF ("Peasant") "Tb^Tf"}
```
Comment (stevecotton) [inline data/multiplayer/scenarios/ANL_utils/ANL_worker_options.cfg:476]: I think this needs to be Uu^Tf. @sevu, please would you confirm?

The corresponding map change isn't in this commit, it was already done in 353dce142ebc55ff954e7de66171b80fb51bfffa. Looks like the effect is that peasants can't harvest mushrooms in 1.15.4.

**Comment by gfgtdf** [comment]
> @gfgtdf I went ahead and deleted the test_lua_mapgen.lua and test_lua_mapgen.cfg files.

I think it'd be better to to do in a seperate  commit though deletinga  a file in a commit named "Mass convert of *^Uf terrain for SP." make it looks like  you accidentally deleted it.

File: data/multiplayer/scenarios/2p_Elensefar_Courtyard.cfg (hunk 2 above)
Comment (Pentarctagon) [reply data/multiplayer/scenarios/2p_Elensefar_Courtyard.cfg]: Done.

**Comment by Pentarctagon** [comment]
Moved the file deletions to a separate commit.

**Comment by gfgtdf** [comment]
The commit message now kinda imples that there used at some point or at least supposed to be used. To make it clear, they were tests, like the unit [test]s for specific features, they were never intended to be used by any other code.

**Comment by Pentarctagon** [comment]
"unused" doesn't have any implication of what they were used for previously.  It means just that - nobody uses them for anything anymore.

File: data/multiplayer/scenarios/ANL_utils/ANL_worker_options.cfg (hunk 3 above)
Comment (sevu) [reply data/multiplayer/scenarios/ANL_utils/ANL_worker_options.cfg:476]: Yes. Easy fix is changing it to `Uu^Tf`. Better fix is using `*^Tf` and using 29 lines farther
[terrain]
x,y=$x1,$y1
terrain="^"
layer=overlay
[/terrain]
@Pentarctagon can change it or remove it from the commit

Comment (Pentarctagon) [reply data/multiplayer/scenarios/ANL_utils/ANL_worker_options.cfg:476]: @sevu You mean changing `{MODIFY_TERRAIN "Uu" $x1 $y1}` to
```
[terrain]
x,y=$x1,$y1
terrain="^"
layer=overlay
[/terrain]
```
?

**Comment by Pentarctagon** [comment]
@nemaara Can you approve, if there's nothing else?

File: data/multiplayer/scenarios/ANL_utils/ANL_worker_options.cfg (hunk 3 above)
Comment (sevu) [reply data/multiplayer/scenarios/ANL_utils/ANL_worker_options.cfg:476]: yes

**Comment by gfgtdf** [comment]
He said if it doesn't change that visuals but it does. That fact that the visuals were already broken in some earlier 1.15.x version is not really important her in my opinion. What matter i that they look different than when they were drawn/written.

**Comment by Pentarctagon** [comment]
I don't follow.

**Comment by Pentarctagon** [comment]
Soliton said that for SP the mushrooms should keep the same look as in 1.14, nemaara agreed, and so I implemented that.

File: data/multiplayer/scenarios/ANL_utils/ANL_worker_options.cfg (hunk 3 above)
Comment (Pentarctagon) [reply data/multiplayer/scenarios/ANL_utils/ANL_worker_options.cfg]: Alright, updated.

**Comment by gfgtdf** [comment]
oh okay my bad then, i just looked at the first file  which happend to be the ai test sceanrio which still converted "Uu^Uf" to "Tb^Tf" so i assumed this pr still changed everything to Tb^Tb

**Comment by Pentarctagon** [comment]
Yeah, stuff like that (the pathfinding unit test as well for example) I kept coverted to Tb^Tf since aesthetics definitely aren't relevant there.

**Review by nemaara — APPROVED** [review]

**Pull request merged**